lxd/containers.go

```diff
@@ -117,6 +96,52 @@ func containersPost(d *Daemon, r *http.Request) Response {
 	return &asyncResponse{run: run, containers: []string{req.Name}}
 }
 
+func createFromNone(d *Daemon, req *containerPostReq) Response {
+	dpath := shared.VarPath("lxc", req.Name)
+	if !shared.PathExists(dpath) {
+		return InternalError(fmt.Errorf("Container path doesn't exist."))
+	}
```

Comment:
On Wed, Feb 25, 2015 at 03:28:55PM -0800, St�phane Graber wrote:
> > @@ -117,6 +96,52 @@ func containersPost(d *Daemon, r *http.Request) Response {
> >  	return &asyncResponse{run: run, containers: []string{req.Name}}
> >  }
> >  
> > +func createFromNone(d *Daemon, req *containerPostReq) Response {
> > +	dpath := shared.VarPath("lxc", req.Name)
> > +	if !shared.PathExists(dpath) {
> > +		return InternalError(fmt.Errorf("Container path doesn't exist."))
> > +	}
> 
> Should that fail though?

I could go either way, really. I can remove it, though.

> It doesn't seem unreasonable to me to have someone define a bunch of containers ahead of time, then going to populate their rootfs before starting them.
> 
> So start definitely should fail but I see no reason to fail at create time.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/lxc/lxd/pull/267/files#r25390851